app/src/main/java/com/asfoundation/wallet/ui/iab/AppcoinsRewardsBuyFragment.kt

```diff
@@ -0,0 +1,232 @@
+    error_message.text = getString(message ?: R.string.activity_iab_error_message)
+    generic_error_layout.visibility = View.VISIBLE
+    hideLoading()
+  }
+
+  override fun finish(uid: String) {
+    presenter.sendPaymentEvent()
+    presenter.sendRevenueEvent()
+    presenter.sendPaymentSuccessEvent()
+    val bundle = billingMessagesMapper.successBundle(uid)
+    bundle.putString(InAppPurchaseInteractor.PRE_SELECTED_PAYMENT_METHOD_KEY,
+        PaymentMethodsView.PaymentMethodId.APPC_CREDITS.id)
+    iabView.finish(bundle)
+  }
+
+  override fun errorClose() {
+    iabView.close(billingMessagesMapper.genericError())
+  }
+
+  override fun finish(purchase: Purchase, orderReference: String?) {
+    presenter.sendPaymentEvent()
+    presenter.sendRevenueEvent()
+    presenter.sendPaymentSuccessEvent()
+    val bundle = billingMessagesMapper.mapPurchase(purchase, orderReference)
+    bundle.putString(InAppPurchaseInteractor.PRE_SELECTED_PAYMENT_METHOD_KEY,
+        PaymentMethodsView.PaymentMethodId.APPC_CREDITS.id)
+    iabView.finish(bundle)
+  }
+
+  override fun showWalletValidation(@StringRes error: Int) {
+    iabView.showWalletValidation(error)
+  }
+
+  override fun showTransactionCompleted() {
+    loading_view.visibility = View.GONE
+    generic_error_layout.visibility = View.GONE
+    iab_activity_transaction_completed.visibility = View.VISIBLE
+  }
+
+  override fun getAnimationDuration(): Long {
+    return lottie_transaction_success.duration
+  }
+
+  override fun lockRotation() {
+    iabView.lockRotation()
+  }
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    check(context is IabView) { "AppcoinsRewardsBuyFragment must be attached to IAB activity" }
+    iabView = context
+  }
+
+  private fun setupTransactionCompleteAnimation() {
+    lottie_transaction_success.setAnimation(R.raw.success_animation)
+  }
+
+  private val amount: BigDecimal by lazy {
+    if (arguments!!.containsKey(AMOUNT_KEY)) {
+      arguments!!.getSerializable(AMOUNT_KEY) as BigDecimal
+    } else {
+      throw IllegalArgumentException("amount data not found")
+    }
+  }
+
+  private val uri: String by lazy {
+    if (arguments!!.containsKey(URI_KEY)) {
+      arguments!!.getString(URI_KEY, "")
+    } else {
+      throw IllegalArgumentException("uri not found")
+    }
+  }
+
+  private val isBds: Boolean by lazy {
+    if (arguments!!.containsKey(IS_BDS)) {
+      arguments!!.getBoolean(IS_BDS)
+    } else {
+      throw IllegalArgumentException("isBds not found")
+    }
+  }
+
+  private val gamificationLevel: Int by lazy {
+    if (arguments!!.containsKey(GAMIFICATION_LEVEL)) {
+      arguments!!.getInt(GAMIFICATION_LEVEL)
+    } else {
+      throw IllegalArgumentException("gamification level data not found")
+    }
+  }
+
+  private val transactionBuilder: TransactionBuilder by lazy {
+    if (arguments!!.containsKey(TRANSACTION_KEY)) {
+      arguments!!.getParcelable(TRANSACTION_KEY) as TransactionBuilder
```

Comment:
My Android Studio complains that this can be null, and it can as on newInstace we are saying that it can be null